Add 2024 Thank You Banners

Abban opened the pull request.

Abban committed:
Add 2024 Thank You Banners

moiikana commented:
Ticket: https://phabricator.wikimedia.org/T382313

moiikana commented:
use `npm run thankyou` for developing

moiikana committed:
Fix height, text highlighting, button text

gbirke committed:
Improve height of thank-you-fullpage

Previous fix did not work correctly on Firefox
Chrome *can* `calc(100ch-80px)`, it was the height that was interfering.
Changed the breakpoint for displaying the fullpage banner centerered
because it was too small to early

Ticket: https://phabricator.wikimedia.org/T382313

gbirke committed:
Make number of members configurable

Number of members displayed in the stats block is now a locale-aware
number that can be changed globally without redeploying the banners.

Added more requirements to cleanup document.

Ticket: https://phabricator.wikimedia.org/T382313

Sperling-0 committed:
Add missing highlight for EN VAR banners

Ticket: https://phabricator.wikimedia.org/T382313

gbirke committed:
Replace tracking placeholder in WPDE TY

Change the regex that matches banners for wikipedia.de to also match
thank you banners for replacing the tracking placeholders in the
compiled source code.

Updated cleanup doc with instructions for regex
Updated cleanup doc with small clarifications that came up while reading it.

Ticket: https://phabricator.wikimedia.org/T382942

moiikana commented:
do not merge until https://phabricator.wikimedia.org/T383110 is done here as well

moiikana committed:
Make cookie distinguishable (#678)

- this allows to differ betweet close cookies caused by thankyou banners vs. usual fundraising banners
- the banner category name on CentralNotice cannot use a '-', it won't work with that symbol

https://phabricator.wikimedia.org/T383110

moiikana commented:
rebased main
